### PR TITLE
refactor: update rexexp match for jira references

### PR DESCRIPTION
**Pull request by ghost**
- Fix regular expression to match entire Jira ticket link in the pull request body.
- Added tests for invalid Jira ticket references in pull request body.
- Fix identified security issue [HIGH]: `Prototype Pollution in JSON5 via Parse Method`: https://github.com/finderau/github-pr-linter/security/dependabot/1

**Security issue summary:**
The parse method of the JSON5 library before and including version 2.2.1 does not restrict parsing of keys named __proto__, allowing specially crafted strings to pollute the prototype of the resulting object.

This vulnerability pollutes the prototype of the object returned by JSON5.parse and not the global Object prototype, which is the commonly understood definition of Prototype Pollution. However, polluting the prototype of a single object can have significant security impact for an application if the object is later used in trusted operations.

### CODE REVIEW

File: src/lint.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ module.exports = function (body, title, failureCallback, infoCallback) {
     } else {
         // Ensure a Jira ticket is referenced on the last line of the body
         infoCallback('Checking PR body Jira reference');
-        const jiraRegex = new RegExp('^(Relates to|Closes) ((?<!([A-Z]{1,10})-?)[A-Z]+-\\d+)$');
+        const jiraRegex = new RegExp('^(Relates to|Closes) (http|https):\/\/finder\.atlassian\.net\/browse\/[A-Za-z]+-[0-9]+$');
```
Comment (lovek323) [inline src/lint.js]: Nice one

**Review by lovek323 — APPROVED** [review]
This is awesome. Thanks @last-marcel

File: src/lint.js (hunk 1 above)
Comment (ghost) [reply src/lint.js]: Required refactor to support Markdown URL syntax.

**Comment by ghost** [comment]
For a detailed breakdown of the regular expression used to match the Jira reference in the PR body, visit: https://regex101.com/r/C35JKl/1

**Pull request merged**